### DISCUSSION TITLE
How to implement a password reset feature using a built-in auth UI.

### DISCUSSION THREAD

**Discussion by HQHAN** [supabase/supabase#26850] [Questions]
I am using a `supabase_auth_ui` for implementing an auth feature for my flutter app.
I have a question regarding a password reset feature.

Currently, `supabase_auth_ui` provides two widgets for the password reset feature, `SupaEmailAuth` and `SupaResetPassword`. I am wondering how to detect a password reset deep link in the app when the user clicks on the password reset link in the email. There is a field called `redirectTo` in `SupaEmailAuth` that I've already assigned the signup complete callback URL to, but I am not sure how to provide a reset password URL there when the UI changes to a password reset form.

Thanks!

```flutter
// Create a Email sign-in/sign-up form
SupaEmailAuth(
  redirectTo: kIsWeb ? null : 'io.mydomain.myapp://callback',
  onSignInComplete: (response) {
    // do something, for example: navigate('home');
  },
  onSignUpComplete: (response) {
    // do something, for example: navigate("wait_for_email");
  },
  metadataFields: [
    MetaDataField(
      prefixIcon: const Icon(Icons.person),
      label: 'Username',
      key: 'username',
      validator: (val) {
        if (val == null || val.isEmpty) {
          return 'Please enter something';
        }
        return null;
      },
    ),
  ],
),
```

```flutter
SupaResetPassword(
  accessToken: supabase.auth.currentSession?.accessToken,
  onSuccess: (UserResponse response) {
    // do something, for example: navigate('home');
  },
  onError: (error) {
    // do something, for example: navigate("wait_for_email");
  },
),
```

**Comment by dshukertjr** [comment]
You can use the `auth.onAuthStateChange` callback to listen to `passwordChange` event to redirect the user to the proper page!

**Reply by dshukertjr** [reply]
Also, I have created a PR after seeing this discussion. I agree that a developer should be able to redirect the user to a different URL for password recover flow. 
https://github.com/supabase-community/flutter-auth-ui/pull/98
